### DISCUSSION TITLE
New Exchange Rates API

### DISCUSSION THREAD

**Discussion by djekl** [ash-jc-allen/laravel-exchange-rates#61] [Ideas]
Hi,

Discussions aren't enabled for this repo, so made an issue. With the breaking change to the [exchangerates.io](https://exchangerates.io) site, and the requirement of a paid plan to set the base rate to something other than EUR etc., there is an alternative API that I found here - [exchangerate.host](https://exchangerate.host)

The base `rates` endpoint looks to be the same, and returns more currencies. However the `historical` endpoint is different (from my initial looks the other day).

If I get some time I will write up a PR unless you, or somebody else, beats me to it. But it's certainly something to look into. Possibly as an option, so users can still use the existing [exchangerates.io](https://exchangerates.io) with a free/paid plan.

**Comment by ash-jc-allen** [comment]
Hi @djekl! Thanks for flagging up the discussions part, I've enabled that now and converted the issue :smile:

I did come across that API when I was making the changes to the package last week. If people are interested in using that API, I would definitely be happy to offer it as an additional service. My original idea was to maybe add a ` driver ` / ` service `/ ` api ` type of field to the config. Then from there, we could use an adapter-y style pattern. By doing this, it'd give us the flexibility to add more services in the future as well.

I haven't had time to sit down and properly think this through, so I'm not entirely sure in what direction I want to take the package. But, please let me know if you do have any ideas or fancy contributing. After all, it's you guys that you are using the package, so I want to make sure that it's built in a way that works for everyone.

**Reply by djekl** [reply]
It certainly is an option. In all honesty I have had your package starred for a while, but we already had something implemented before finding this. So for me it was as simple as swapping out the uri to get us working again. Other than that I also haven't had the time to dig into it. But I do think that it's something worth exploring if more people are interested.

**Reply by ash-jc-allen** [reply]
Yeah 100%, I think this could be a really good option that might encourage people to use the package a little bit more (and maybe move over to it). I probably won't have much time to take a look at this for a few weeks, but if you (or anyone else reading this in the future) fancy making any PRs, I'd be more than happy to take a look :smile:

**Comment by theissn** [comment]
Happy to help on this issue too (change came as a pretty big surprise and didn't notice till sentry started flagging errors lol), the XML API that the European Central Bank maintains might also be a good addition. I believe that they are similar though need to have a proper look

https://www.ecb.europa.eu/stats/policy_and_exchange_rates/euro_reference_exchange_rates/html/index.en.html

**Reply by ash-jc-allen** [reply]
Thanks, @theissn I appreciate the offer to help! Maybe in the coming weekend, I'll try and get something planned out and get it added as an issue/milestone to work towards.

I haven't worked with any XML APIs in the past, but I think that'd be a useful API to have seeing as we're going straight to the source. I think a lot of the other APIs are just wrappers around the European Central Bank API.

If you come up with any ideas before the weekend, feel free to fire them over to us :smile:
